fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVCache.java

```diff
@@ -64,11 +64,8 @@ public static MTMVCache from(MTMV mtmv, ConnectContext connectContext) {
         Plan originPlan = planner.plan(unboundMvPlan, PhysicalProperties.ANY, ExplainLevel.REWRITTEN_PLAN);
         Plan mvPlan = originPlan.accept(new DefaultPlanRewriter<Object>() {
             @Override
-            public Plan visit(Plan plan, Object context) {
-                if (plan instanceof LogicalResultSink) {
-                    return ((LogicalResultSink<Plan>) plan).child();
-                }
-                return super.visit(plan, context);
+            public Plan visitLogicalResultSink(LogicalResultSink<? extends Plan> logicalResultSink, Object context) {
+                return logicalResultSink.child().accept(this, context);
```

Comment:
file sink need keep order too. so u need override visitLogicalFileSink

Reply:
this code is aim to remove the resultSink for query rewrite by materilized view later. Remove sort under result sink is  in another pr